Source/Dafny/Compiler-java.cs

```diff
@@ -2923,58 +2932,63 @@ private class GenericArrayElementLvalue : ILvalue {
-        } else {
-          wrEquals.WriteLine("return true;");
-        }
+      var wrEquals = wr.NewBlock("public boolean equals(Object obj)");
+      wrEquals.WriteLine("if (this == obj) return true;");
+      wrEquals.WriteLine("if (obj == null) return false;");
+      wrEquals.WriteLine("if (getClass() != obj.getClass()) return false;");
+      wrEquals.WriteLine($"Tuple{i} o = (Tuple{i}) obj;");
+      if (i != 0) {
+        wrEquals.WriteLine("return {0};", Util.Comma(" && ", i, j => $"java.util.Objects.equals(this._{j}, o._{j})"));
+      }
+      else {
+        wrEquals.WriteLine("return true;");
       }
 
       wr.WriteLine();
       wr.WriteLine("@Override");
-      using (var wrToString = wr.NewBlock("public String toString()")) {
-        wrToString.WriteLine("StringBuilder sb = new StringBuilder();");
-        wrToString.WriteLine("sb.append(\"(\");");
-        for (int j = 0; j < i; j++) {
-          wrToString.WriteLine($"sb.append(_{j} == null ? \"null\" : _{j}.toString());");
-          if (j != i - 1)
-            wrToString.WriteLine("sb.append(\", \");");
-        }
-        wrToString.WriteLine("sb.append(\")\");");
-        wrToString.WriteLine("return sb.toString();");
+      var wrToString = wr.NewBlock("public String toString()");
+      wrToString.WriteLine("StringBuilder sb = new StringBuilder();");
+      wrToString.WriteLine("sb.append(\"(\");");
+      for (int j = 0; j < i; j++)
+      {
+        wrToString.WriteLine($"sb.append(_{j} == null ? \"null\" : _{j}.toString());");
+        if (j != i - 1)
+          wrToString.WriteLine("sb.append(\", \");");
       }
 
+      wrToString.WriteLine("sb.append(\")\");");
+      wrToString.WriteLine("return sb.toString();");
+
       wr.WriteLine();
       wr.WriteLine("@Override");
-      using (var wrHashCode = wr.NewBlock("public int hashCode()")) {
-        wrHashCode.WriteLine("// GetHashCode method (Uses the djb2 algorithm)");
-        wrHashCode.WriteLine("// https://stackoverflow.com/questions/1579721/why-are-5381-and-33-so-important-in-the-djb2-algorithm");
-        wrHashCode.WriteLine("long hash = 5381;");
-        wrHashCode.WriteLine("hash = ((hash << 5) + hash) + 0;");  // this is constructor 0 (in fact, it's the only constructor)
-        for (int j = 0; j < i; j++) {
-          wrHashCode.WriteLine("hash = ((hash << 5) + hash) + java.util.Objects.hashCode(this._" + j + ");");
-        }
-        wrHashCode.WriteLine("return (int)hash;");
+      var wrHashCode = wr.NewBlock("public int hashCode()");
+      wrHashCode.WriteLine("// GetHashCode method (Uses the djb2 algorithm)");
+      wrHashCode.WriteLine(
+        "// https://stackoverflow.com/questions/1579721/why-are-5381-and-33-so-important-in-the-djb2-algorithm");
+      wrHashCode.WriteLine("long hash = 5381;");
+      wrHashCode.WriteLine(
+        "hash = ((hash << 5) + hash) + 0;"); // this is constructor 0 (in fact, it's the only constructor)
+      for (int j = 0; j < i; j++)
+      {
```

Comment:
I think the style in the Dafny code base is to put the open `{` on the same line as the `for` or `foreach` keyword (see, e.g., the previous L2965 above). This PR uses a different style for these, but I suggest not changing the style.

Reply:
I've gone through all the compilers files to make sure the right style is used.